src/preload.py

```diff
@@ -955,8 +956,20 @@ def get_partition(name, image=None):
 if __name__ == "__main__":
     update_ca_certificates(**SH_OPTS)
     for line in sys.stdin:
-        data = json.loads(line)
-        method = methods[data["command"]]
-        result = method(**data.get("parameters", {}))
-        print(json.dumps({"result": result}))
+        try:
+            data = json.loads(line)
+            method = methods[data["command"]]
+            result = method(**data.get("parameters", {}))
+            response = {
+                "result": result,
+                "statusCode": 0,
+            }
+        except BaseException:
+            response = {
+                "error": traceback.format_exc(),
+                "statusCode": 1,
+            }
+        print(json.dumps(response))
         sys.stdout.flush()
+        if response["statusCode"]:
+            break
```

Comment:
This `break` means that the Python process should exit when the first error is encountered, rather than continue running and waiting for the next command on `stdin`. I thought this makes sense because, on the CLI / `preload.js` side, I am desperately trying to ensure that the CLI process exits on detecting any errors (to solve the poor UX of infinite spinning wheels).